### PR TITLE
Catch exceptions

### PR DESCRIPTION
**Pull request by kirstenmg**
When the user causes an error, they now get a semi-specific error message and the exception is not printed to the console (unless DEBUG is set to True). Please try to cause some errors and see if they are caught.

Note that this branch also includes the save option.

### CODE REVIEW

**Comment by CLiu13** [comment]
@kirstenmg will need you to resolve merge conflicts first before I can review - if you scroll down to the bottom of the page, you should see a button that says "Resolve conflicts." click on that and it will walk through the areas that need to be fixed 👍

**Comment by kirstenmg** [comment]
I've resolved the merge conflicts!

File: main.py [hunk 1]
```diff
@@ -3,15 +3,21 @@
 import matplotlib.animation as animation
 import random
 import datetime
+import traceback
 from tkinter import *
 from tkinter import tix, filedialog, font, messagebox
 
+# determine whether to print caught exceptions
+DEBUG = False
```
Comment (CLiu13) [inline main.py:11]: Good idea! I wonder if we could add this as a checkbox on the UI side, so users wouldn't have to dig through the code to enable this if they wanted to? (up to y'all as usual)

**Comment by CLiu13** [comment]
@linasc3-ai @puria123 reviews needed 🙂

**Pull request merged**